### PR TITLE
[#11086] Enrolling students: show a progress indicator when enrolling a large class

### LINKED ISSUE #11086: Enrolling students: show a progress indicator when enrolling a large class
v7.13.0

When a large class is being enrolled in one go, perhaps we should show some progress indicator? Currently, we only show a spinner which can cause the user to think the enrolling is stuck.

### PR DESCRIPTION
**Pull request by FergusMok**
Fixes #11086 

Added a progress bar to the student enrollment. 

Example with 225 students for enrollment: 
https://user-images.githubusercontent.com/10182564/155832625-dd2e993a-7260-4152-bfd2-7c2a5986306f.mp4

### CODE REVIEW

**Review by ziqing26 — APPROVED** [review]
LGTM!

**Comment by moziliar** [comment]
Reminder to resolve the conflicts!

**Review by NicolasCwy — APPROVED** [review]
LGTM!

**Review by NicolasCwy — APPROVED** [review]
LGTM!

File: src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.ts [hunk 1]
```diff
@@ -188,9 +189,8 @@ export class InstructorCourseEnrollPageComponent implements OnInit {
         }),
     );
 
-    enrollRequests.pipe(finalize(() => {
-      this.isEnrolling = false;
-    })).subscribe({
```
Comment (wkurniawan07) [inline src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.ts:193]: Unrelated change here. The only change that should be observed here is the addition of `this.progressBarService.updateProgress(0);`.

File: src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.html [hunk 2]
```diff
@@ -96,14 +96,16 @@ <h1 id="enroll-header" class="text-break">Enroll Students for {{ courseId }}</h1
...
                     type="submit"
                     title="Enroll"
                     id="btn-enroll"
                     name="button-enroll"
                     class="btn btn-success float-right"
                     [disabled]="isEnrolling"
                     (click)="submitEnrollData()">
-                  <tm-ajax-loading *ngIf="isEnrolling"></tm-ajax-loading>
+                    <tm-progress-bar *ngIf="isEnrolling"></tm-progress-bar>
+                    <tm-ajax-loading *ngIf="isEnrolling"></tm-ajax-loading>
```
Comment (wkurniawan07) [inline src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.html:108]: Unrelated change and inconsistent indentation (compare with the `Enroll students` line after this)

**Review by wkurniawan07 — CHANGES_REQUESTED** [review]

**Review by wkurniawan07 — APPROVED** [review]

**Pull request merged**